api_v3/services/DataService.php

```diff
@@ -187,4 +187,52 @@ function serveAction($entryId, $version = -1, $forceProxy = false)
 			}
 		}	
 	}
+
+
+	/**
+	* Update the dataContent of data entry using a resource
+	*
+	* @action addContent
+	* @param string $entryId
+	* @param KalturaDataEntryAllowedContentResource $resource
+	* @return string
+	* @throws KalturaErrors::ENTRY_ID_NOT_FOUND
+	* @validateUser entry entryId edit
+	*/
+	function addContentAction($entryId, KalturaDataEntryAllowedContentResource $resource)
+	{
+		$dbEntry = entryPeer::retrieveByPK($entryId);
+
+		if (!$dbEntry || $dbEntry->getType() != KalturaEntryType::DATA)
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
+
+		if($resource) {
```

Comment:
NO need to check if resource , since it is from dedicated type and cannot be null